### ISSUE TITLE
FluffyChat does not install the actual software

### ISSUE THREAD

**Issue by Starz0r** [A-d-r-i/update_choco_package#12]
Looks like the FluffyChat package is downloading the source code instead of pointing to the installer, e.g: [here](https://gitlab.com/api/v4/projects/16112282/packages/generic/fluffychat/1.6.3/fluffychat-windows.msix). I can open a PR with the proper change if you'd like.

**Comment by A-d-r-i** [comment]
Thanks for your feedback!
It's a good idea indeed, a direct installer will save me steps in the installation but:

- Unfortunately I don't think that's possible at the moment, the [Chocolatey roadmap](https://docs.chocolatey.org/en-us/roadmap#chocolatey-cli-choco) indicates that the .MSIX installers will be integrated in the future.

> the source code instead of pointing to the installer

- It's not the source code directly, because the executable is already compiled inside. Normally there is no difference between a classic installation (.EXE/.MSI which uncompresses) and an installation from this asset there (.ZIP). When I "built" my code to send the package, the asset with .MSIX didn't exist, that's why it uses the zipped folder directly.

I therefore do not prefer to change the method for the moment by taking these elements into account. If you ever encounter a problem with the installation of the package related to the fact that it installs from a compressed folder, do not hesitate to let me know! 😊

**Comment by A-d-r-i** [comment]
Furthermore:

> FluffyChat does not install the actual software

It's wrong, it does install the latest version of Fluffychat because I'm downloading the asset directly from the latest stable version : [send_fluffychat.ps1#L23](https://github.com/A-d-r-i/update_choco_package/blob/main/send_fluffychat.ps1#L23)

**Comment by Starz0r** [comment]
> It's wrong, it does install the latest version of Fluffychat because I'm downloading the asset directly from the latest stable version : [send_fluffychat.ps1#L23](https://github.com/A-d-r-i/update_choco_package/blob/main/send_fluffychat.ps1?rgh-link-date=2022-09-08T16%3A08%3A11Z#L23)

This does not look to be true, it seems to be downloading the top artifact from Gitlab (which is always the source code zip in this case). Downloading from https://gitlab.com/famedly/fluffychat/-/archive/v1.6.4/fluffychat-v1.6.4.zip (the download URL in the nupkg), shows that there aren't any executable files there, just the source.

> Unfortunately I don't think that's possible at the moment, the [Chocolatey roadmap](https://docs.chocolatey.org/en-us/roadmap#chocolatey-cli-choco) indicates that the .MSIX installers will be integrated in the future.

I'm currently managing 3 different packages that uses MSIX/APPX installers, all the documentation says is that they don't have "direct support". You can use extension packages or invoke the system's Package Installer to properly install these packages. I'd always recommend using the MSIX/APPX installer over any portable ZIP, which should be done in a `.portable` metapackage.

**Comment by A-d-r-i** [comment]
> shows that there aren't any executable files there, just the source.

You're right! Sorry for the misunderstanding, the download url was wrong. It's [fixed](https://github.com/A-d-r-i/update_choco_package/blob/main/send_fluffychat.ps1#L23) and I've pushed a [fix version](https://community.chocolatey.org/packages/fluffychat/1.6.4.20220909) to chocolatey. Everything works now.

> I'm currently managing 3 different packages that uses MSIX/APPX installers

I would like to see an example if possible. After some tests I have the impression that it depends on the windows version for installation with .MSIX

**Comment by Starz0r** [comment]
> You're right! Sorry for the misunderstanding, the download url was wrong. It's [fixed](https://github.com/A-d-r-i/update_choco_package/blob/main/send_fluffychat.ps1?rgh-link-date=2022-09-09T06%3A48%3A46Z#L23) and I've pushed a [fix version](https://community.chocolatey.org/packages/fluffychat/1.6.4.20220909) to chocolatey. Everything works now.

Great! The application appears to install correctly now.

> I would like to see an example if possible. After some tests I have the impression that it depends on the windows version for installation with .MSIX

Not sure what's up with the documentation, but we've had `Add-AppxPackage` built in since Chocolatey 0.9. Keep in mind, with MSIX/MSIXbundle/APPX, you do need to check if it's already installed like so:
```ps1
if ((Get-AppxPackage -name $AppxPackageName).Version -Match $Version) {
  if($env:ChocolateyForce) {
	Get-AppxPackage -Name $AppxPackageName | Remove-AppxPackage;
  } else {
	return;
  }
}```, plus some logging to the screen for users to see which code paths were hit, and why, and if they're overridable, and announcing that all to the user as well.

I will always recommend using MSIX over portable zips, as the underlying system manages them better, and has over all native support. Although some users may not want that, which is why if you do this, you should also break the zip-based installation into its own `.portable` metapackage. Gives users more choice over what type of installation they may want, but the MSIX installation should be default.
